### PR TITLE
Change private methods to protected in rules

### LINKED ISSUE #44: Make rules extendable
I would like to extend some of the rules, like UnusedLocalVariable and others without resorting to copy and paste. It would be much easier if those classes had protected methods instead of private methods. That way I can implement a specific ruleset for my projects, but still get any bug fixes or enhancements to those rules as they become available.

[Synced from PivotalTracker: 'http://www.pivotaltracker.com/story/show/9355955']

### PR DESCRIPTION
**Pull request by artem-prozorov**
Type: **refactoring**
Issue: Resolves #44 
Breaking change: no

The changes are pretty simple and straightforward. No tests are needed.

### CODE REVIEW

File: src/main/php/PHPMD/Rule/UnusedFormalParameter.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ class UnusedFormalParameter extends AbstractLocalVariable implements FunctionAwa
      *
      * @var \PHPMD\Node\ASTNode[]
      */
-    private $nodes = array();
+    protected $nodes = [];
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Rule/UnusedFormalParameter.php:35]: This change is not related to this PR and would require a change of the PHP defined minimum level. So please remove it.

File: src/main/php/PHPMD/Rule/UnusedFormalParameter.php [hunk 2]
```diff
@@ -60,7 +60,7 @@ public function apply(AbstractNode $node)
             return;
         }
 
-        $this->nodes = array();
+        $this->nodes = [];
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Rule/UnusedFormalParameter.php:63]: This change is not related to this PR and would require a change of the PHP defined minimum level. So please remove it.

File: src/main/php/PHPMD/Rule/UnusedFormalParameter.php [hunk 3]
```diff
@@ -243,13 +243,13 @@ private function removeCompoundVariables(AbstractNode $node)
...
      */
-    private function removeVariablesUsedByFuncGetArgs(AbstractNode $node)
+    protected function removeVariablesUsedByFuncGetArgs(AbstractNode $node)
     {
         $functionCalls = $node->findChildrenOfType('FunctionPostfix');
 
         foreach ($functionCalls as $functionCall) {
             if ($this->isFunctionNameEqual($functionCall, 'func_get_args')) {
-                $this->nodes = array();
+                $this->nodes = [];
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/Rule/UnusedFormalParameter.php:252]: This change is not related to this PR and would require a change of the PHP defined minimum level. So please remove it.

File: src/main/php/PHPMD/Rule/UnusedFormalParameter.php (hunk 1 above)
Comment (artem-prozorov) [reply src/main/php/PHPMD/Rule/UnusedFormalParameter.php]: Sorry, I haven't thought of this from keeping backward compatibility point of view

Restored original form

File: src/main/php/PHPMD/Rule/UnusedFormalParameter.php (hunk 2 above)
Comment (artem-prozorov) [reply src/main/php/PHPMD/Rule/UnusedFormalParameter.php]: done

File: src/main/php/PHPMD/Rule/UnusedFormalParameter.php (hunk 3 above)
Comment (artem-prozorov) [reply src/main/php/PHPMD/Rule/UnusedFormalParameter.php]: done
